### PR TITLE
CRuby Unit 2 Lesson 3: Producing and Consuming Avro with a Schema Registry

### LINKED ISSUE #28: Ruby Unit 2, Lesson 3: Producing and Consuming Avro with a Schema Registry

### PR DESCRIPTION
**Pull request by aviflax**
Closes #28.

### CODE REVIEW

File: ruby/cruby/unit_2/docker-compose.yml [hunk 1]
```diff
@@ -2,24 +2,33 @@ version: '3.5'
...
+    depends_on: [zookeeper]
+    expose: [9092]
+    networks: [unit]
     environment:
       KAFKA_BROKER_ID: 1
       KAFKA_ZOOKEEPER_CONNECT: 'zookeeper:2181'
       KAFKA_ADVERTISED_LISTENERS: 'PLAINTEXT://kafka:9092'
-      KAFKA_OFFSETS_TOPIC_REPLICATION_FACTOR: 1
+      KAFKA_OFFSETS_TOPIC_REPLICATION_FACTOR: '1'
+      CONFLUENT_SUPPORT_METRICS_ENABLE: 'false'
```
Comment (fussybeaver) [inline ruby/cruby/unit_2/docker-compose.yml]: Is this environment variable used in the guides ?

File: ruby/cruby/unit_2/lesson_3/TimeZone.avsc [hunk 2]
```diff
@@ -0,0 +1,28 @@
+{
+  "type": "record",
+  "name": "TimeZone",
+  "fields": [
+    {
+      "name": "country_code",
+      "type": "string",
+      "doc": "In the tz database this is named `cc`."
```
Comment (fussybeaver) [inline ruby/cruby/unit_2/lesson_3/TimeZone.avsc]: Is this doc string relevant to the tutorial ? I'm not quite sure what the `tz` database is ...

File: ruby/cruby/unit_2/docker-compose.yml (hunk 1 above)
Comment (aviflax) [reply ruby/cruby/unit_2/docker-compose.yml]: If you mean the lessons, then… sort-of. Indirectly. This just disables the plugin or whatever that ships with the CP distribution of the Kafka broker that sends anonymous “support” metrics to Confluent. It’s probably not truly necessary, but I like turning off features that we don’t need because they can be distracting (log output) and could possibly even lead to some kind of failure (e.g. if the Internet is unreachable who knows what happens).

File: ruby/cruby/unit_2/lesson_3/TimeZone.avsc (hunk 2 above)
Comment (aviflax) [reply ruby/cruby/unit_2/lesson_3/TimeZone.avsc]: > Is this doc string relevant to the tutorial ?

No, it’s not super relevant. We could drop it. I’m not 100% sure why I included it… I guess I really like the `doc` property of Avro schemata; I really appreciate that it exists and is supported, and I guess I wanted to publicize it a bit, if subtly.

Do you think I should remove it?

>  I'm not quite sure what the `tz` database is ...

Description from [the Wikipedia article](https://en.wikipedia.org/wiki/Tz_database):

> The tz database is a collaborative compilation of information about the world's time zones, primarily intended for use with computer programs and operating systems. Paul Eggert is its current editor and maintainer, with the organizational backing of ICANN. The tz database is also known as tzdata, the zoneinfo database or IANA time zone database, and occasionally as the Olson database, referring to the founding contributor, Arthur David Olson.
